components/SearchTrackers.vue

```diff
@@ -0,0 +1,121 @@
+        resetActive(element : HTMLElement) {
+          let add : Boolean = !element.classList.contains('active');
+          this.clearActive();
+          if(add){
+            element.classList.add('active');
+          }
+        },
+        /**
+         * gets called whenever the active element of the search list must be cleared
+         * and does this by looping over all 'active' elements
+         */
+        clearActive(){
+        let currentActive : HTMLCollectionOf<Element> = document.getElementsByClassName("active");
+          for (var i = 0; i < currentActive.length; i++) {
+            currentActive.item(i)?.classList.remove('active');
+          }
+        }
+    },
+    mounted(){
+      /**
+       * event triggered on closing the details window, sent from the DetailsPane component
+       */
+      this.$root?.$on("details-closed", ()=> {
+        this.clearActive();
+      })
+    },
+    data(){
+      return {
+        /**
+         * example trackers list, where the value will be all relevant info for that tracker
+         * There is no need for these value lists to be equal size, some can have more/less pairs
+         */
+        trackers: {
+          'Alpha 01' : {"description": "Description of Alpha 01","nickname": "Florian"},
+          'Alpha 02' : {"description": "Description of Alpha 02","nickname": "Bo"},
+          'Alpha 03' : {"description": "Description of Alpha 03","nickname": "Ben"},
+          'Alpha 04' : {"description": "Description of Alpha 04","nickname": "Joshua"},
+          'Phone 01' : {"description": "Description of Phone 01","nickname": "Meg"},
+          'Phone 02' : {"description": "Description of Phone 02","nickname": "Jonathan", "groups" : "NCCN, flooding,  kitten stuck in a tree"},
+        }
+      }
+    }
+})
+</script>
+<template>
+      <div class="container" text-variant="light">
+        <!-- search box row-->
+        <div class="row align-items-center form-group">
+          <div class="col-sm-2"></div>
+          <div class="col-sm-8">
+            <b-input-group>
+              <b-form-input size="sm" placeholder="search tracker"></b-form-input>
+              <b-input-group-append>
+                <b-button size="sm" variant="outline-success" @click="performSearch">🔍</b-button>
+              </b-input-group-append>
+            </b-input-group>
+          </div>
+          <div class="col-sm-2"></div>
+        </div>
+        <!-- tracker data row-->
+        <div class="row align-items-center align-text-center">
+          <div class="col-sm-2"></div>
+          <div class="col-sm-8">
+            <ul class="list-group" id="trackerList">
+              <li class="list-group-item" v-for="(item, index) in trackers" :key="index" @click="clickListItem($event, item)" :id="index">
+                {{ index }}
+              </li>
+            </ul>
+          </div>
+          <div class="col-sm-2"></div>
+        </div>
+      </div>
+</template>
+
+<style lang="scss">
```

Comment:
Can we centralize all scss code into another folder ?

Reply:
We could, but the idea was to use scoped styling (so one component's style does not interfere with others). Therefor, I think keeping the style separate for each component is a good idea.

Reply:
I agree with @JoshuaMoelans with this one. While we could look into this when we're working in big projects that repeat something very often, Vue is not designed to handle external files being loaded on it *(But it is possible)*.

This should also give you the urge to split maybe files up and this is sometime you should do, as you should strive to your components being **as atomic as possible** to increase maintainability.